### ISSUE TITLE
500 Internal Server Error

### ISSUE THREAD

**Issue by JOE-DCG** [hfaran/slack-export-viewer#96]
Installed the newest Python for Windows on 2 machines. Both failing with the 500 Internal Server Error again when running slack-export-viewer.

IndexError: list index out of range

**Comment by hfaran** [comment]
Please post the full stacktrace. It's impossible to debug with information this limited.

**Comment by k-spit** [comment]
I ran into the same issue as @JOE-DCG. 

`ubuntu@ubuntu:~/slack-export$ slack-export-viewer -z slack_export.zip/tmp/_slackviewer/6081d0fe198607ec79321dde75e9aca4f2546d77 already exists
 * Serving Flask app "slackviewer.app" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://localhost:5000/ (Press CTRL+C to quit)
127.0.0.1 - - [21/Feb/2019 11:39:53] "GET / HTTP/1.1" 500 -
Error on request:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 270, in run_wsgi
    execute(self.server.app)
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 258, in execute
    application_iter = app(environ, start_response)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python2.7/dist-packages/slackviewer/app.py", line 81, in index
    return channel_name(channels[0])
IndexError: list index out of range`

**Comment by hfaran** [comment]
@k-spit Two options to help debug here:

1) Please paste/upload the contents of your channels.json from the archive here so I can take a look, or
2) Add a breakpoint here to see what's happening when your channels are attempted to be loaded

**Comment by k-spit** [comment]
Here is the content of my channels.json from the archive. Because it is a company account, I had to anonymize values. Hope this does not matter.

`[
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  },
  {
    "id": "anon",
    "name": "anon",
    "is_channel": "anon",
    "created": "anon",
    "is_archived": "anon",
    "is_general": "anon",
    "unlinked": "anon",
    "creator": "anon",
    "name_normalized": "anon",
    "is_shared": "anon",
    "is_org_shared": "anon",
    "is_member": "anon",
    "is_private": "anon",
    "is_mpim": "anon",
    "members": "anon",
    "topic": "anon",
    "purpose": "anon",
    "previous_names": "anon",
    "num_members": "anon"
  }
]`

**Comment by karlding** [comment]
I bisected this, and it seems like this is due to the change introduced in #93.

```
 → TITANIC@slack-export-viewer ((43fe2aa...)) $ git checkout 0.8.2
Previous HEAD position was 43fe2aa add missing import, fixes regression in reaction handling (#92)
HEAD is now at 54f99f5 Merge branch 'master' of github.com:hfaran/slack-export-viewer
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((0.8.2)) $ git bisect start
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((0.8.2)|BISECTING) $ git bisect good
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((0.8.2)|BISECTING) $ git checkout master
Previous HEAD position was 54f99f5 Merge branch 'master' of github.com:hfaran/slack-export-viewer
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
(slack-export-viewer)
 → TITANIC@slack-export-viewer (master|BISECTING) $ git bisect bad
Bisecting: 3 revisions left to test after this (roughly 2 steps)
[16a4895126c8465912b9053f9ed4872b88389eb5] add aditional @channel formatter
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((16a4895...)|BISECTING) $ git bisect good
Bisecting: 1 revision left to test after this (roughly 1 step)
[f746a77f419b2a76e625763a52fc0fa0aa5faba8] convert user id and emoji to proper html content
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((f746a77...)|BISECTING) $ git bisect bad
Bisecting: 0 revisions left to test after this (roughly 0 steps)
[57d25d708ebb34ef4ab0e983773f30f938172e0d] show emoji bare name test like :+1:
(slack-export-viewer)
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((57d25d7...)|BISECTING) $ git bisect good
f746a77f419b2a76e625763a52fc0fa0aa5faba8 is the first bad commit
commit f746a77f419b2a76e625763a52fc0fa0aa5faba8
Author: Lukas Kahwe Smith <smith@pooteeweet.org>
Date:   Fri Jan 4 18:43:54 2019 +0100

    convert user id and emoji to proper html content
```

If I revert this commit (`f746a77f419b2a76e625763a52fc0fa0aa5faba8`) it works as expected.

The problem is because `channels` is always an instance of a `str` due to the way the argument is passed in.

**Comment by karlding** [comment]
I bisected this, and it seems like this is due to the change introduced in #93.

```
 → TITANIC@slack-export-viewer (master) $ git co 22944173e2d87d3b8f9f71924f1ed2bc3abaa944
Note: checking out '22944173e2d87d3b8f9f71924f1ed2bc3abaa944'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 2294417 Add release instructions
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((2294417...)) $ git bisect start
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((2294417...)|BISECTING) $ git bisect good
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((2294417...)|BISECTING) $ git co master
Previous HEAD position was 2294417 Add release instructions
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
(slack-export-viewer)
 → TITANIC@slack-export-viewer (master|BISECTING) $ git bisect bad
Bisecting: 2 revisions left to test after this (roughly 2 steps)
[57d25d708ebb34ef4ab0e983773f30f938172e0d] show emoji bare name test like :+1:
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((57d25d7...)|BISECTING) $ git bisect good
Bisecting: 0 revisions left to test after this (roughly 1 step)
[43fe2aa490393d1be459bc032009e49d318e5f85] add missing import, fixes regression in reaction handling (#92)
(slack-export-viewer)
 → TITANIC@slack-export-viewer ((43fe2aa...)|BISECTING) $ git bisect good
90ca642551d8d907125ab4a09f7e97bfc1dfd72d is the first bad commit
commit 90ca642551d8d907125ab4a09f7e97bfc1dfd72d
Author: Lukas Kahwe Smith <smith@pooteeweet.org>
Date:   Tue Jan 15 21:44:07 2019 +0100

    added 'channels', 'no-sidebar' and 'no-external-references' options (#93)

:100644 100644 3aa0e30d52d70c5c52ef279e3f3d5c1c1e9c4bfa 3847d58c271e7c17c44edd696374af66a23bf042 M      README.md
:040000 040000 ea48496e7465cdccdef4f1741dbc10dde005bbf5 3c773b20916615f025f0736443ecdee66e159621 M      slackviewer
```

If I revert this commit (`90ca642551d8d907125ab4a09f7e97bfc1dfd72d`) it works as expected.

The problem is because `channels` is always an instance of a `str` due to the way the argument is passed in.

**Comment by hfaran** [comment]
@karlding to confirm, your new PRs fix this issue, correct?

**Comment by karlding** [comment]
Yes, this should be fixed by #98.
